fedimint-testing/src/btc/mock.rs

```diff
@@ -222,18 +225,26 @@ impl BitcoinTest for FakeBitcoinTest {
             ref mut txid_to_block_height,
             ..
         } = *inner;
+        let block_height = blocks.len() as u64;
         FakeBitcoinTest::mine_block(addresses, blocks, pending, txid_to_block_height);
         let block_header = inner.blocks.last().unwrap().header;
         let proof = TxOutProof {
             block_header,
             merkle_proof,
         };
+
         inner
             .proofs
             .insert(transaction.compute_txid(), proof.clone());
-        inner
-            .scripts
-            .insert(address.script_pubkey(), vec![transaction.clone()]);
+
+        if let Some(txs) = inner.scripts.get_mut(&address.script_pubkey()) {
```

Comment:
Nit: I think it can be done cleaner with `inner.scripts.entry(address.script_pubkey()).or_default().push((transaction.clone(), block_height));